pkg/email/template.go

```diff
@@ -0,0 +1,75 @@
+package email
+
+import (
+	"html/template"
+	"net/url"
+	"strings"
+)
+
+const (
+	ResetPasswordPath = "/auth/resetpassword" //#nosec
+	InviteUserPath    = "/auth/users/create"  //#nosec
+
+	ResetPasswordEmailSubject = "Reset Password Request for your LakeFS account"
+	InviteUserWEmailSubject   = "You have been invited to LakeFS"
+
+	inviteUserTemplate = `Hello, <br>
+You have been invited to join LakeFS <br>
+Click on this link to set up your account <br>
+{{.URL}} <br>
+Thanks <br>
+The LakeFS team <br>
+`
+
+	resetEmailTemplate = `Hello, <br>
+A request has been received to change the password to your account,  <br>
+Click on this link to reset your password <br>
+{{.URL}} <br>
+If you did not initiate this request you can please disregard this email. <br>
+Thanks  <br>
+The LakeFS team <br>
+	`
+)
+
+var (
+	templ              = template.New("template")
+	ResetEmailTemplate = template.Must(templ.Parse(resetEmailTemplate))
+	InviteUserTemplate = template.Must(templ.Parse(inviteUserTemplate))
+)
+
+type Link struct {
+	URL string
+}
+
+func buildURL(host string, path string, qParams map[string]string) (*string, error) {
+	baseURL, err := url.Parse(host)
```

Comment:
Need to add verification code on configuration load to verify that we will not fail here at runtime